streamalert/alert_processor/outputs/credentials/provider.py

```diff
@@ -399,16 +399,28 @@ def save_credentials(self, descriptor, credentials, kms_key_alias):
             else credentials.data()
         )
 
-        parameter_name = self.get_parameter_name(descriptor)
+        parameter_name = self._get_parameter_name(descriptor)
 
         return AwsSsm.put_parameter(parameter_name, unencrypted_creds, self._region, kms_key_alias)
 
-    def get_parameter_prefix(self):
-        """Generate the parameter prefix from prefix and descriptor"""
-        return "{}_streamalert_secrets".format(self._prefix)
+    def _get_parameter_name(self, descriptor):
+        """
+        Returns the name of the AWS SSM Parameter Store parameter in which the current output
+        service stores its credentials.
+
+        (!) NOTE: The leading forward slash character is intentional for parameters that are
+            in a hierarchy.
+
+        @see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services
+                /ssm.html#SSM.Client.put_parameter
 
-    def get_parameter_name(self, descriptor):
-        parameter_prefix = self.get_parameter_prefix()
+        Params:
+            descriptor (str): Descriptor of current output
+
+        Returns:
+            str
+        """
+        parameter_prefix = "{}/streamalert/outputs".format(self._prefix)
```

Comment:
instead of the aggressive `(!) NOTE`, couldn't we just do this below...

```python
parameter_suffix = get_formatted_output_credentials_name(self._service_name, descriptor)
return '/{}/streamalert/outputs/{}'.format(prefix, parameter_suffix)
```

Reply:
also double quotes (`"`) vs single quotes (`'`)